### PR TITLE
[ASA-367][SM]: Overseas Agents - Create 'company registration number'…

### PR DESCRIPTION
**Pull request by sureshhmrc**
… page

### CODE REVIEW

File: app/uk/gov/hmrc/agentoverseasapplicationfrontend/controllers/ApplicationController.scala [hunk 1]
```diff
@@ -158,10 +158,27 @@ class ApplicationController @Inject()(
...
   }
+
+  def submitCompanyRegistrationNumber: Action[AnyContent] = validApplicantAction.async { implicit request =>
+    withAgentSession { session =>
+      CompanyRegistrationNumberForm.form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Ok(company_registration_number(formWithErrors)),
+          validFormValue => {
+            sessionStoreService
```
[the remark this answers is not in the document]
Comment (sureshhmrc) [reply app/uk/gov/hmrc/agentoverseasapplicationfrontend/controllers/ApplicationController.scala]: fixed

File: it/uk/gov/hmrc/agentoverseasapplicationfrontend/controllers/ApplicationControllerISpec.scala [hunk 2]
```diff
@@ -552,4 +552,49 @@ class ApplicationControllerISpec extends BaseISpec {
...
+        .withFormUrlEncodedBody("confirmRegistration" -> "true", "registrationNumber" -> "ABC123")
+
+      val result = await(controller.submitCompanyRegistrationNumber(authenticatedRequest))
+      status(result) shouldBe 303
+
+      result.header.headers(LOCATION) shouldBe routes.ApplicationController.showTaxRegistrationNumberForm().url
+      await(sessionStoreService.fetchAgentSession).get.companyRegistrationNumber shouldBe Some(CompanyRegistrationNumber(Some(true), Some("ABC123")))
+    }
+
+    "show validation error if no choice was selected" in {
```
Comment (arkowalski) [inline it/uk/gov/hmrc/agentoverseasapplicationfrontend/controllers/ApplicationControllerISpec.scala]: need another one of this testing for empty reg number

File: it/uk/gov/hmrc/agentoverseasapplicationfrontend/controllers/ApplicationControllerISpec.scala [hunk 3]
```diff
@@ -552,4 +552,49 @@ class ApplicationControllerISpec extends BaseISpec {
       result should containMessages("taxRegNo.form.no-radio.selected")
     }
   }
+
+  "GET /company-registration-number" should {
+    "display the company-registration-number form" in {
```
Comment (arkowalski) [inline it/uk/gov/hmrc/agentoverseasapplicationfrontend/controllers/ApplicationControllerISpec.scala]: should check for the back button too, both cases if ukTaxRegistered= Yes and Not/Unsure cases

File: it/uk/gov/hmrc/agentoverseasapplicationfrontend/controllers/ApplicationControllerISpec.scala (hunk 2 above)
Comment (sureshhmrc) [reply it/uk/gov/hmrc/agentoverseasapplicationfrontend/controllers/ApplicationControllerISpec.scala]: fixed
